src/drivers/hmc5883/hmc5883.cpp

```diff
@@ -1347,13 +1347,14 @@ start_bus(struct hmc5883_bus_option &bus, enum Rotation rotation)
 	}
 			
 	int fd = open(bus.devpath, O_RDONLY);
-	if (fd < 0)
+	if (fd != -1) {
 		return false;
+	}
 
 	if (ioctl(fd, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_DEFAULT) < 0) {
-		close(fd);
```

Comment:
we still need this close(), as errx() exits (although if nuttx guarantees closing all FDs on exit as posix does then we're OK)